src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -13,26 +13,36 @@ namespace Microsoft.AspNetCore.Components;
 public class PersistentComponentState
 {
     private IDictionary<string, byte[]>? _existingState;
-    private readonly IDictionary<string, byte[]> _currentState;
 
-    private readonly List<Func<Task>> _registeredCallbacks;
+    private readonly IDictionary<string, byte[]> _currentServerState;
+    private readonly IDictionary<string, byte[]> _currentWebAssemblyState;
+
+    private readonly List<Func<Task>> _registeredServerCallbacks;
+    private readonly List<Func<Task>> _registeredWebAssemblyCallbacks;
+
+    private readonly ISerializationModeHandler _serializationModeHandler;
 
     internal PersistentComponentState(
-        IDictionary<string, byte[]> currentState,
-        List<Func<Task>> pauseCallbacks)
+        IDictionary<string, byte[]> currentServerState,
+        IDictionary<string, byte[]> currentWebAssemblyState,
+        List<Func<Task>> pauseServerCallbacks,
+        List<Func<Task>> pauseWebAssemblyCallbacks,
+        ISerializationModeHandler serializationModeHandler)
     {
-        _currentState = currentState;
-        _registeredCallbacks = pauseCallbacks;
+        _currentServerState = currentServerState;
+        _currentWebAssemblyState = currentWebAssemblyState;
+        _registeredServerCallbacks = pauseServerCallbacks;
+        _registeredWebAssemblyCallbacks = pauseWebAssemblyCallbacks;
+        _serializationModeHandler = serializationModeHandler;
     }
 
     internal bool PersistingState { get; set; }
 
+    internal PersistedStateSerializationMode CurrentSerializationMode { get; set; } = PersistedStateSerializationMode.Infer;
```

Comment:
Why is the default "infer"?

Reply:
It doesn't need to be. I'll remove this. Thanks for noting!